common/events.cpp

```diff
@@ -73,15 +72,18 @@ get_uuid()
 
 int EventPublisher::init(const string event_source)
 {
+    int rc = -1;
     m_zmq_ctx = zmq_ctx_new();
-    void *sock = zmq_socket (m_zmq_ctx, ZMQ_PUB);
 
-    int rc = zmq_connect (sock, get_config(XSUB_END_KEY).c_str());
-    RET_ON_ERR(rc == 0, "Publisher fails to connect %s", get_config(XSUB_END_KEY).c_str());
+    void *sock = zmq_socket (m_zmq_ctx, ZMQ_PUB);
+    RET_ON_ERR(sock != NULL, "Failed to ZMQ_PUB socket");
 
     rc = zmq_setsockopt (sock, ZMQ_LINGER, &LINGER_TIMEOUT, sizeof (LINGER_TIMEOUT));
```

Comment:
No logical code change except 
1) added null check for sock
2) moved linger timeout to happen before connect.